packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -320,21 +321,17 @@ export default class ConfigParser {
      */
     setFilePathToFilterOptions(cliArgFileList: string[], specs: Spec[]) {
         const filesToFilter = new Set<string>()
-        const fileList = ConfigParser.getFilePaths(specs, this._config.rootDir, this._pathService)
+        const fileList = this.findSpecs(specs)
         cliArgFileList.forEach(filteredFile => {
             filteredFile = removeLineNumbers(filteredFile)
             // Send single file/file glob to getFilePaths - not supporting hierarchy in spec/exclude
             // Return value will always be string[]
-            const globMatchedFiles = <string[]>ConfigParser.getFilePaths(
-                this._pathService.glob(filteredFile, path.dirname(this.#configFilePath)),
-                this._config.rootDir,
-                this._pathService
-            )
+            const globMatchedFiles = <string[]> this.findSpecs(this._pathService.glob(filteredFile, this.#projectRootDir))
             if (this._pathService.isFile(filteredFile)) {
                 filesToFilter.add(
                     this._pathService.ensureAbsolutePath(
                         filteredFile,
-                        path.dirname(this.#configFilePath)
+                        this.#projectRootDir
```

Comment:
files should always resolve based on the location of the config file not the root path